### PR TITLE
feat: add nuxt-auth

### PR DESCRIPTION
**Pull request by BracketJohn**
Add nuxt-auth to the modules overview.

I'm a bit anxious about the name and would like to get your opinion on this @Atinux / ... -> `auth` already exists as [the nuxt 2 auth module](https://github.com/nuxt-community/auth-module) with an availability for nuxt 2 only. Do you think it's ok to add `nuxt-auth` as a separate thing here, with it's availability targeted for nuxt 3 only? I think that may be clear enough in the overview, but would love to get your perspective here.

Feel free to hit me up on discord at BracketJohn or on twitter via DM if you want to discuss in a more sync-matter. 

Thanks!

### CODE REVIEW

**Comment by Atinux** [comment]
Made a PR to improve readme: https://github.com/sidebase/nuxt-auth/pull/15

I want to know more about the bundle size cost about the module before advocating: https://github.com/sidebase/nuxt-auth/issues/2#issuecomment-1296847974

Also, what is happening for the user to blink between page navigations?


https://user-images.githubusercontent.com/904724/200806712-5d55c070-fdc6-4c34-aef9-af251b032b63.mp4

We cannot push a module to the list without being a miminum rigorous in term of DX, UX and performance.

Anyway I really appreciate what you guys are doing for the Nuxt community with Sidebase 🚀

**Comment by BracketJohn** [comment]
> Made a PR to improve readme: https://github.com/sidebase/nuxt-auth/pull/15

Thanks! I reviewed it over there.

> I want to know more about the bundle size cost about the module before advocating: https://github.com/sidebase/nuxt-auth/issues/2#issuecomment-1296847974

Yes, I agree that this should happen before we merge this PR + release `0.1.0` (aka. come out of beta). Is there a bundle size you would consider the limit, or is it rather about adding the right warnings, e.g., to the readme, if the bundle size is bigger than X? I can already say that if the bundle size is significantly increased by `nuxt-auth` it will be a priority + issue anyways to cut it.

> Also, what is happening for the user to blink between page navigations?

I'm certain that this is because `useSession` is re-called on the sub-pages. We use `useState` to ensure sharing between components & pages without a required re-fetch, but this still needs to wait for some loading to finish:

https://github.com/sidebase/nuxt-auth/blob/23f1ac0be850f8cde1fd3fc4a155270f0ce42d21/src/runtime/composables/useSession.ts#L70-L71

A mitigation would be to to initially set the state to `loading` in every case, before accessing `useState`. Do you have something better in mind that can avoid the initial loading-flash?

**Comment by BracketJohn** [comment]
> I want to know more about the bundle size cost about the module before advocating: https://github.com/sidebase/nuxt-auth/issues/2#issuecomment-1296847974

Update: I've done a first, naive benchmark, check it out in here: https://github.com/sidebase/nuxt-auth/issues/2#issuecomment-1308990378

TLDR, Overview of results:
image

I would suggest that we discuss in the issue though @Atinux

**Comment by BracketJohn** [comment]
Hey @Atinux - any indication at what point you'll get around to 
- further your docs PR (minor review from me there),
- have a look at my size benchmark and give your verdict?

Would be great to have this done before nuxt itself becomes stable (:

**Comment by Atinux** [comment]
@BracketJohn after discussing with the team, could you rename the file & image to `sidebase-auth` please so we can move forward?

**Comment by BracketJohn** [comment]
@Atinux thanks for getting back on this -> I've renamed the files as requested.

**Review by Atinux — APPROVED** [review]
